### ISSUE TITLE
Additional ngspice and Spectre simulation support

### ISSUE THREAD

**Issue by rohanku** [ucb-substrate/substrate2#322]
- [ ] Add support for additional simulator-specific flags
- [ ] Add support for additional analysis types
- [ ] Add support for additional simulation primitives

**Comment by kcaisley** [comment]
I'm working on this now. I see that you some refactoring in #347, moving `Tran` under the `spectre::analysis` module and adding the `Montecarlo` simulation type. 

I'd like to create an `Op` DC operating point with support for variable sweeps, and add support for options like transient noise to the existing `Tran` under `spectre::analysis::tran`. Then, I will make a unit test creating some inverters, and simulating their jitter and DC transfer characteristic.

Regarding your point

> Add support for additional simulation primitives

Does this essentially equate to extending the built-in "schema", as discussed [here?](https://docs.substratelabs.io/docs/schematics/scir#schemas-1)

**Comment by rohanku** [comment]
Yes, that is correct. One thing to note, we are now using https://github.com/rahulk29/psfparser for parsing Spectre output (though we are still using `nutlex` for `ngspice`). If you are interested in working on Spectre, you will need to update the parser to parse additional analysis outputs.
